controller/cmd/destination/main.go

```diff
@@ -48,18 +48,24 @@ func Main(args []string) {
 	}
 
 	global, err := config.Global(consts.MountPathGlobalConfig)
-	if err != nil {
-		log.Fatalf("Failed to load global config: %s", err)
-	}
 
 	trustDomain := ""
 	if *disableIdentity {
 		log.Info("Identity is disabled")
 	} else {
-		trustDomain = global.GetIdentityContext().GetTrustDomain()
+		if err != nil {
+			trustDomain = global.GetIdentityContext().GetTrustDomain()
+		} else {
+			trustDomain = "cluster.local"
+			log.Warnf("failed to load trust domain from global config: [%s] (falling back to %s)", err, trustDomain)
+		}
```

Comment:
switch `err != nil` check.

going a step further to match the `clusterDomain` block below:
```go
trustDomain = global.GetIdentityContext().GetTrustDomain()
if err != nil || trustDomain == "" {
	trustDomain = "cluster.local"
	log.Warnf("failed to load trust domain from global config: [%s] (falling back to %s)", err, trustDomain)
}
```

Reply:
good catch, thanks